### PR TITLE
complete CreateKey_napi 、Encrypt_napi、Decrypt_napi、GenerateDataKey_napi

### PR DESCRIPTION
**Pull request by Liang1XZhao**
The content includes nodejs and C + +

### CODE REVIEW

File: core/App/ehsm_napi.cpp [hunk 1]
```diff
@@ -52,50 +58,53 @@ static char* StringToChar(std::string str)
...
-    void addData(std::string key, std::string data)
-    {
+	
+	void setCode(int code){code = code;};
+	void setMessage(string message){msg = message;};
+	void addData(string key, uint32_t data) {
+        if(jsonStr.size() > 0){
+            jsonStr += ",";
+        }
+        jsonStr += "\""+key+"\" : " + "\""+intToString(data)+"\"";
```
Comment (syan10) [inline core/App/ehsm_napi.cpp:76]: seem you didn't modify intToString to std::to_string()

File: core/App/ehsm_napi.cpp [hunk 2]
```diff
@@ -272,41 +315,145 @@ char* NAPI_Decrypt(const char* cmk,
...
+        message: string,
+        result: {
+            plaintext_base64 : string,
+            ciphertext_base64 : string,
+        }
+    }
+*/
+char* NAPI_GenerateDataKey(const char* cmk_base64,
+        const char* aad)
+{
```
Comment (syan10) [inline core/App/ehsm_napi.cpp:366]: seems this API are defined wrong!!
please see the definitions in the header file:
https://github.com/intel/ehsm/blob/main/core/App/ehsm_napi.h#L90
char* NAPI_GenerateDataKey(const char* cmk,
        const uint32_t keylen,
        const char* aad);

Comment (syan10) [reply core/App/ehsm_napi.cpp:366]: you have missed the argument "keylen"...

File: core/App/ehsm_napi.h [hunk 3]
```diff
@@ -84,11 +105,17 @@ char* NAPI_AsymmetricDecrypt(const char* cmk,
...
+        code: int,
+        message: string,
+        result: {
+            plaintext_base64 : string,
+            cipherText_base64 : string,
+        }
+    }
 */
-char* NAPI_GenerateDataKey(const char* cmk,
-        const uint32_t keylen,
```
Comment (syan10) [inline core/App/ehsm_napi.h:91]: why delete the keylen?

File: core/App/ehsm_napi.cpp (hunk 1 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp:76]: It will be fix.

File: ehsm_kms_service/ehsm_kms_server.js [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+       code: int,
+       message: string,
+       result: {
+         plaintext_base64 : string,
+         cipherText_base64 : string,
+       }
+    }
+   */
+  'NAPI_GenerateDataKey': ['string',['string','string']],
+
```
Comment (syan10) [inline ehsm_kms_service/ehsm_kms_server.js:187]: the backend API NAPI_GenerateDataKey lost a parameter "keylen"

File: ehsm_kms_service/ehsm_kms_params.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  EH_INTERNAL_KEY: 0,
+  EXTERNAL_KEY: 1,
+}
+
+const _6kb_length = 6 * 1024;
+
+const aad = {
+  type: 'string',
+  maxLength: 32,
+  length: 100
```
Comment (syan10) [inline ehsm_kms_service/ehsm_kms_params.js:22]: what's the meaning about length: 100?

File: ehsm_kms_service/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "ehsm kms service",
+  "main": "ehsm_kms_server.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node ehsm_kms_server.js"
+  },
+  "keywords": [
+    "ehsm"
+  ],
+  "author": "neusoft",
```
Comment (syan10) [inline ehsm_kms_service/package.json:13]: author change it to "intel"

File: ehsm_kms_service/package.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  "license": "ISC",
```
Comment (syan10) [inline ehsm_kms_service/package.json:14]: what's the meaning abou this license?

File: ehsm_kms_service/test/test_aes.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
+import requests 
+import json
+
+# Replace <port> with real port number
+port = '3000'
+base_url = "http://127.0.0.1:" + port + "/ehsm?Action="
```
Comment (syan10) [inline ehsm_kms_service/test/test_aes.py:6]: why not set another macro for the addr?
ehsm_srv_addr = 'http://127.0.0.1'

base_url =  ehsm_srv_addr +":"+ port + "/ehsm?Action="

File: core/App/ehsm_napi.cpp (hunk 2 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp:366]: It will be fix.

File: core/App/ehsm_napi.h (hunk 3 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.h:91]: It will be fix.

File: ehsm_kms_service/ehsm_kms_server.js (hunk 4 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_server.js:187]: It will be fix.

File: core/App/ehsm_napi.cpp [hunk 9]
```diff
@@ -252,14 +288,21 @@ char* NAPI_Decrypt(const char* cmk,
...
+    decode_cmk = base64_decode(cmk_base64);
 
-    decode_cipher = base64_decode(ciphertext);
+    decode_cipher = base64_decode(ciphertext_base64);
 
     ret = ehsm_deserialize_cmk(&masterkey, (const uint8_t*)decode_cmk.data(), decode_cmk.size());
     if (ret != EH_OK) {
-        retJsonObj.code = retJsonObj.CODE_FAILED;
-        retJsonObj.msg = "Failed to deserialize cmk";
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
```
Comment (syan10) [inline core/App/ehsm_napi.cpp:304]: can we set a meaningful errorcode? e.g: retJsonObj.CODE_FAILED+ret
retJsonObj.setCode(retJsonObj.CODE_FAILED);

it's will be helpful to debug the error.

File: ehsm_kms_service/ehsm_kms_params.js (hunk 5 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_params.js:22]: It will be fix.

File: ehsm_kms_service/package.json (hunk 6 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/package.json:13]: It will be fix.

File: core/App/ehsm_napi.cpp (hunk 9 above)
Comment (syan10) [reply core/App/ehsm_napi.cpp:304]: same with below

File: ehsm_kms_service/package.json (hunk 7 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/package.json:14]: It will be fix.

File: ehsm_kms_service/test/test_aes.py (hunk 8 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_aes.py:6]: It will be fix.

File: core/App/ehsm_napi.cpp (hunk 9 above)
Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp]: It will be fix.

**Pull request closed without merging**